dash/include/dash/allocator/CollectiveAllocator.h

```diff
@@ -123,12 +126,15 @@ class CollectiveAllocator
   /**
    * Move-assignment operator.
    */
-  self_t & operator=(const self_t && other) noexcept
+  self_t & operator=(self_t && other) noexcept
   {
     // Take ownership of other instance's allocation vector:
     if (this != &other) {
-      std::swap(_allocated, other._allocated);
```

Comment:
Ok, got your point: We have to differentiate between destruction of the allocator, deallocation of the vector and deallocation of the `dart_gptr`. I implemented it in this way to be absolutely sure that the vector is empty as otherwise in [`clear()`](https://github.com/dash-project/dash/blob/20fdf5d3d5799681c145504f589f611673fa6d82/dash/include/dash/allocator/LocalAllocator.h#L226) each `dart_gptr` is freed. This function is called in the deallocator of `LocalAllocator` and hence would invalidate the `dart_gptr` of the new `LocalAllocator`.